sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/PartitionKey.java

```diff
@@ -75,4 +75,10 @@ public boolean equals(Object other) {
         PartitionKey otherKey = Utils.as(other, PartitionKey.class);
         return otherKey != null && this.internalPartitionKey.equals(otherKey.internalPartitionKey);
     }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode();
```

Comment:
super.hashCode() will call Object's hashcode() API, which should not be the case here. 
You are breaking the hashcode() contract here. This can lead to very serious bugs when partitionKey is used in any Collections like Lists, Sets, Maps, etc. 
The hashcode contract: https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode--

In my opinion, the correct implementation should be this: 
```
import java.util.Objects;


...


@Override
    public int hashCode() {
        return Objects.hash(internalPartitionKey);
    }
```

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/6804